src/main/java/driver/Parser.java

```diff
@@ -0,0 +1,57 @@
+package driver;
+
+import exception.DukeException;
+import exception.IncompleteInputException;
+import command.Command;
+import command.ListCommand;
+import command.DoneCommand;
+import command.DeleteCommand;
+import command.AddCommand;
+import command.SearchCommand;
+
+/**
+ *
+ */
+
+public class Parser {
+
+    /**
+     *
+     */
+
+    public Command parse(String userInput) throws DukeException, IncompleteInputException {
```

Comment:
Good method for parsing! However, do you think that you could break down the pieces of code in each if-else statement and make them into separate methods? I feel that would help you improve readability and cleanliness of your code.